tooling/nargo/src/ops/fuzz.rs

```diff
@@ -0,0 +1,349 @@
+                                root_path: root_path.clone(),
+                                package_name: package_name.clone(),
+                            }
+                            .build_with_base(base)
+                        };
+                        #[cfg(not(feature = "rpc"))]
+                        let build_foreign_call_executor = |output, base| {
+                            DefaultForeignCallBuilder { output, enable_mocks: true }
+                                .build_with_base(base)
+                        };
+                        // Use a base layer that doesn't handle anything, which we handle in the `execute` below.
+                        let inner_executor = build_foreign_call_executor(
+                            if show_output { PrintOutput::Stdout } else { PrintOutput::None },
+                            layers::Unhandled,
+                        );
+
+                        let mut foreign_call_executor =
+                            TestForeignCallExecutor::new(inner_executor);
+
+                        execute_program_with_brillig_fuzzing(
+                            program,
+                            initial_witness,
+                            &B::default(),
+                            &mut foreign_call_executor,
+                            Some(location_to_feature_map),
+                        )
+                        .map_err(|(nargo_err, brillig_coverage)| {
+                            (
+                                nargo_err.to_string()
+                                    + ": "
+                                    + &nargo_err
+                                        .user_defined_failure_message(&brillig_error_types)
+                                        .unwrap_or("<no message>".to_owned()),
+                                brillig_coverage,
+                            )
+                        })
+                    };
+                let abi = acir_program.abi.clone();
+                let acir_and_brillig_programs = AcirAndBrilligPrograms {
+                    acir_program: acir_program.into(),
+                    brillig_program: brillig_program.into(),
+                };
+                let mut fuzzer = FuzzedExecutor::new(
+                    acir_and_brillig_programs,
+                    acir_executor,
+                    brillig_executor,
+                    &package_name.clone().unwrap(),
+                    context.def_interner.function_name(&fuzzing_harness.get_id()),
+                    FuzzedExecutorExecutionConfiguration {
+                        num_threads: fuzz_execution_config.num_threads,
+                        timeout: fuzz_execution_config.timeout,
+                    },
+                    FuzzedExecutorFailureConfiguration {
+                        fail_on_specific_asserts: fuzzing_harness.only_fail_enabled(),
+                        failure_reason: fuzzing_harness.failure_reason(),
```

Comment:
It seems that if `failure_reason()` is `Some` then `only_fail_enabled()` will be `true`, so carrying these two things is a bit redundant. This could be simplified to just carry `failure_reason` (dropping the bool).

Reply:
I'm going to add should_fail_with compatibility to be able to run tests on greybox fuzzing next. So bools will be needed